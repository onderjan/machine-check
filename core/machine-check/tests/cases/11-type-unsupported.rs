#[machine_check_macros::machine_description]
mod machine {
    struct A {
        a: ::machine_check::UnsupportedType,
        b: [ArrayElementType; 1],
        c: &mut u32,
    }

    impl A {
        fn a() -> ::machine_check::UnsupportedType {}
        fn b() -> [ArrayElementType; 2] {}
        fn c() -> &mut u32 {}

        fn d(&mut self) -> u32 {}

        fn e(a: ::machine_check::UnsupportedType) -> u32 {}
        fn f(a: [ArrayElementType; 1]) -> u32 {}
        fn g(a: &mut u32) -> u32 {}

        fn h() -> u32 {
            let a: ::machine_check::UnsupportedType;
            let b: [ArrayElementType; 1];
            let c: &mut u32;

            let d: <A as B>::Q;

            0
        }
    }
}

fn main() {}
