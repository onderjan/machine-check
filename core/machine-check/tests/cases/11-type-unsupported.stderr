error: machine-check: Unknown machine-check type
 --> tests/cases/11-type-unsupported.rs:4:12
  |
4 |         a: ::machine_check::UnsupportedType,
  |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: machine-check: Non-path type not supported
 --> tests/cases/11-type-unsupported.rs:5:12
  |
5 |         b: [ArrayElementType; 1],
  |            ^^^^^^^^^^^^^^^^^^^^^

error: machine-check: Non-path type not supported
 --> tests/cases/11-type-unsupported.rs:6:12
  |
6 |         c: &mut u32,
  |            ^^^^^^^^

error: machine-check: Unknown machine-check type
  --> tests/cases/11-type-unsupported.rs:10:19
   |
10 |         fn a() -> ::machine_check::UnsupportedType {}
   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: machine-check: Non-path type not supported
  --> tests/cases/11-type-unsupported.rs:11:19
   |
11 |         fn b() -> [ArrayElementType; 2] {}
   |                   ^^^^^^^^^^^^^^^^^^^^^

error: machine-check: Non-path type not supported
  --> tests/cases/11-type-unsupported.rs:12:19
   |
12 |         fn c() -> &mut u32 {}
   |                   ^^^^^^^^

error: machine-check: Mutable receiver argument not supported
  --> tests/cases/11-type-unsupported.rs:14:15
   |
14 |         fn d(&mut self) -> u32 {}
   |               ^^^

error: machine-check: Unknown machine-check type
  --> tests/cases/11-type-unsupported.rs:16:17
   |
16 |         fn e(a: ::machine_check::UnsupportedType) -> u32 {}
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: machine-check: Non-path type not supported
  --> tests/cases/11-type-unsupported.rs:17:17
   |
17 |         fn f(a: [ArrayElementType; 1]) -> u32 {}
   |                 ^^^^^^^^^^^^^^^^^^^^^

error: machine-check: Mutable references not supported
  --> tests/cases/11-type-unsupported.rs:18:18
   |
18 |         fn g(a: &mut u32) -> u32 {}
   |                  ^^^

error: machine-check: Unknown machine-check type
  --> tests/cases/11-type-unsupported.rs:21:20
   |
21 |             let a: ::machine_check::UnsupportedType;
   |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: machine-check: Non-path type not supported
  --> tests/cases/11-type-unsupported.rs:22:20
   |
22 |             let b: [ArrayElementType; 1];
   |                    ^^^^^^^^^^^^^^^^^^^^^

error: machine-check: Mutable references not supported
  --> tests/cases/11-type-unsupported.rs:23:21
   |
23 |             let c: &mut u32;
   |                     ^^^

error: machine-check: Quantified self not supported
  --> tests/cases/11-type-unsupported.rs:25:20
   |
25 |             let d: <A as B>::Q;
   |                    ^^^^^^^^^^^
